server/src/main/java/org/opensearch/cluster/coordination/CoordinationState.java

```diff
@@ -459,6 +461,10 @@ public PublishResponse handlePublishRequest(PublishRequest publishRequest) {
             clusterState.term()
         );
         persistedStateRegistry.getPersistedState(PersistedStateType.LOCAL).setLastAcceptedState(clusterState);
+        if (isRemoteStateEnabled && publishRequest.getManifest() != null && localNode.isClusterManagerNode()) {
```

Comment:
Should we check for publication enabled as well? or isRemoteStateEnabled has implicit check for it?

Reply:
I thought as remote state is enabled, even if publication might be disabled, master re-election should not be triggering full cluster state upload. But since we are not downloading the cluster state from the manifest in those cases, we don't have manifest in our `PublishRequest`, and we don't end up inside this if check.